inbm/trtl/factory/compose_test.go

```diff
@@ -11,17 +11,33 @@ var mockedExitStatus = 0
 var mockedStdout string
 var mockedStderr string
 
-func Test_is_username_registry_safe_pass(t *testing.T) {
-     result := is_username_registry_safe("user_name.", "server-name:")
-     assert.Equal(t, true, result)
+func TestReturnNilRegistrySafe(t *testing.T) {
+	assert.Equal(t, isRegistryCredentialsSafe("username", "servername"), nil)}
+
+func TestReturnErrorUserNameTooLong(t *testing.T) {
+	expected := errors.New("error: docker registry username can not exceed 30 characters")
```

Comment:
**<h2>$\color{#FFD700}Coding Style$</h2>**
**Suggestion:** Ensure consistent use of error messages, specifically regarding the character limit error message.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
 expected := errors.New("error: docker registry username cannot exceed 30 characters")
```